desc/compute/_fast_ion.py

```diff
@@ -354,3 +359,354 @@ def fun(pitch_inv):
         surf_batch_size,
     )
     return data
+
+
+@register_compute_fun(
+    name="Jpar",
+    label=(
+        # J_∥ = ∫ dl v_∥ /(√E ∫ dl)
+        "\\J_{\\parallel} = \\integrate v_{\\parallel} dl/\\integrate dl/B"
+    ),
+    units="~",
+    units_long="~",
+    description="Normalized second adiabatic invariant of motion.",
+    coordinates="r",
+    dim=1,
+    profiles=[],
+    params=[],
+    transforms={"grid": []},
+    data=[
+        "min_tz |B|",
+        "max_tz |B|",
+        "B^zeta",
+    ]
+    + Bounce2D.required_names,
+    resolution_requirement="tz",
+    grid_requirement={"can_fft2": True},
+    **_bounce_doc,
+)
+@partial(
+    jit,
+    static_argnames=[
+        "Y_B",
+        "num_transit",
+        "num_well",
+        "num_quad",
+        "num_pitch",
+        "pitch_batch_size",
+        "surf_batch_size",
+        "spline",
+    ],
+)
+def _Jpar(params, transforms, profiles, data, **kwargs):
+    """Second adiabatic invariant of particle motion.
+
+    The normalization requires a length for which we have uesd the fieldline
+    length ∫ dl/B.
+    Typically calculated as a function of (rho, alpha, lambda)
+    """
+    # noqa: unused dependency
+    theta = kwargs["theta"]
+    Y_B = kwargs.get("Y_B", theta.shape[-1] * 2)
+    alpha = kwargs.get("alpha", jnp.array([0.0]))
+    num_transit = kwargs.get("num_transit", 20)
+    num_pitch = kwargs.get("num_pitch", 64)
+    num_well = kwargs.get("num_well", Y_B * num_transit)
+    pitch_batch_size = kwargs.get("pitch_batch_size", None)
+    surf_batch_size = kwargs.get("surf_batch_size", 1)
+    assert (
+        surf_batch_size == 1 or pitch_batch_size is None
+    ), f"Expected pitch_batch_size to be None, got {pitch_batch_size}."
+    spline = kwargs.get("spline", True)
+    fl_quad = (
+        kwargs["fieldline_quad"] if "fieldline_quad" in kwargs else leggauss(Y_B // 2)
+    )
+    quad = (
+        kwargs["quad"]
+        if "quad" in kwargs
+        else get_quadrature(
+            leggauss(kwargs.get("num_quad", 32)),
+            (automorphism_sin, grad_automorphism_sin),
```

Comment:
My rule of thumb is to use `chebgauss2` instead of `leggauss` for weakly singular integrals.